nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java

```diff
@@ -72,6 +77,27 @@ public AbstractReportingTaskNode(final ReportingTask reportingTask, final String
         this.reportingTask = reportingTask;
         this.processScheduler = processScheduler;
         this.serviceLookup = controllerServiceProvider;
+
+        final Class<?> reportingClass = reportingTask.getClass();
+
+        try {
+            if (reportingClass.isAnnotationPresent(DefaultSchedule.class)) {
+                DefaultSchedule dsc = reportingClass.getAnnotation(DefaultSchedule.class);
```

Comment:
I am wondering if we should instead use Spring's AnnotationUtils and its discovery operations there. Basically the question is if you *only* looking for annotation in the specific class _reportingTask.getClass()_, then the above is OK i guess, but I am afraid that if impl of the ReportingTask has a more complex hierarchy and annotated operation is in some super class, then the above will not work. WDYT?